uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -726,135 +757,112 @@ define([
         vm.period = {};
         // inform user if absence period is not found
         vm.loading['fromDayTypes'] = false;
+
         return $q.reject('Please change date as it is not in any absence period');
       }
 
-      return $q.resolve(true);
+      return $q.resolve(previousAbsencePeriodId !== vm.period.id);
     }
 
-    /**
-     * Converts given date to javascript date as expected by uib-datepicker
-     *
-     * @param {String} date from server
-     * @return {Date}
-     */
-    function _convertDateFormatFromServer (date) {
-      return moment(date, sharedSettings.serverDateFormat).toDate();
-    }
-
-    /**
-     * Converts given date to server format
-     *
-     * @param {Date} date
-     * @return {String} date converted to server format
-     */
-    function _convertDateToServerFormat (date) {
-      return moment(date).format(sharedSettings.serverDateFormat);
-    }
-
-    /**
-     * Initializes user's calendar (work patterns)
-     *
-     * @return {Promise}
-     */
-    function _loadCalendar () {
-      return Calendar.get(vm.request.contact_id, vm.period.start_date, vm.period.end_date)
-        .then(function (usersCalendar) {
-          vm.calendar = usersCalendar;
-        });
-    }
-
-    /**
-     * Sets dates (and times in case of "hours" absence type) for vm.request from UI
-     */
-    function _setDates () {
+    function setDatesToRequest () {
```

Comment:
no documentation provided

Reply:
Provided documentation, removed unnecessary code that is already used in other function.